apps/evm/ui/pool/CreateSection/CreateSectionReviewModalTrident.tsx

```diff
@@ -168,122 +175,119 @@ export const CreateSectionReviewModalTrident: FC<CreateSectionReviewModalTrident
-          if (input1.currency.isNative) {
-            value = input1.quotient.toString()
-          }
+        liquidityInput.push({
+          token: input0.currency.isNative ? zeroAddress : (input0.currency.wrapped.address as Address),
+          native: true,
+          amount: BigInt(input0.quotient.toString()),
+        })
+      }
 
-          liquidityInput.push({
-            token: input1.currency.isNative ? AddressZero : input1.currency.wrapped.address,
-            native: true,
-            amount: input1.quotient.toString(),
-          })
+      if (input1) {
+        if (input1.currency.isNative) {
+          value = input1.quotient
         }
 
-        setRequest({
-          from: address,
-          to: contract.address,
-          data: batchAction({
-            contract,
-            actions: [
-              approveMasterContractAction({
-                router: contract,
-                signature: signature,
-              }),
-              deployNewPoolAction({
-                assets: [input0.currency, input1.currency],
-                factory: factory.address,
-                router: contract,
-                feeTier: fee,
-                twap: false,
-              }),
-              getAsEncodedAction({
-                contract,
-                fn: 'addLiquidity',
-                args: [
-                  liquidityInput,
-                  poolAddress,
+        liquidityInput.push({
+          token: input1.currency.isNative ? zeroAddress : (input1.currency.wrapped.address as Address),
+          native: true,
+          amount: BigInt(input1.quotient.toString()),
+        })
+      }
+
+      return {
+        from: address,
+        to: contract.address,
+        data: batchAction({
+          actions: [
+            approveMasterContractAction({
+              signature: signature,
+            }),
+            deployNewPoolAction({
+              assets: [input0.currency, input1.currency],
+              factory: factory.address,
+              feeTier: fee,
+              twap: false,
+            }),
+            encodeFunctionData({
+              ...contract,
+              functionName: 'addLiquidity',
+              args: [
+                liquidityInput,
+                poolAddress,
+                BigInt(
                   pool
                     .getLiquidityMinted(
                       totalSupply,
                       input0.wrapped.toShare(totals?.[token0.wrapped.address]),
                       input1.wrapped.toShare(totals?.[token1.wrapped.address])
                     )
-                    .quotient.toString(),
-                  encoded,
-                ],
-              }),
-            ],
-          }),
-          ...(value && { value }),
-        })
-      } catch (e: unknown) {
-        //
+                    .quotient.toString()
+                ),
+                encoded,
+              ],
+            }),
+          ],
+        }),
+        ...(value && { value }),
```

Comment:
Better to do { value: value? value : undefined }?

Reply:
Don't think it matters, will fix though